engine/consensus/matching/engine.go

```diff
@@ -487,28 +500,27 @@ func (e *Engine) sealableResults() ([]*flow.ExecutionResult, error) {
 			return nil, fmt.Errorf("could not retrieve block: %w", err)
 		}
 
-		// ensure that previous result is known.
-		// get the previous result from our mempool or storage
-
-		previousID := result.PreviousResultID
-
-		// look for previous in mempool and storage
-		previous, found := e.results.ByID(previousID)
-		if !found {
+		// look for previous result in mempool and storage
+		var previous *flow.ExecutionResult
+		previousID := incorporatedResult.Result.PreviousResultID
+		previousResults := e.incorporatedResults.ByResultID(previousID)
+		if previousResults == nil || len(previousResults) == 0 {
 			var err error
 			previous, err = e.resultsDB.ByID(previousID)
 			if errors.Is(err, storage.ErrNotFound) {
-				log.Debug().Msg("skipping result with unknown previous result")
+				log.Debug().Msg("skipping sealable result with unknown previous result")
 				continue
 			}
 			if err != nil {
 				return nil, fmt.Errorf("could not get previous result: %w", err)
 			}
+		} else {
+			previous = previousResults[0].Result
 		}
 
 		// check sub-graph
 		if block.ParentID != previous.BlockID {
-			_ = e.results.Rem(result.ID())
+			_ = e.incorporatedResults.Rem(incorporatedResult.ID())
```

Comment:
I found the name `IncorporatedResults` is a bit confusing. 

At beginning, I thought all results in the `IncorporatedResults` mempool has passed the check and guaranteed its parent result has already been seen (either exists in the mempool, or in the storage). 

But actually it's not the case. What we did here, is to add a result to a mempool first, then performace checks to remove results that are not incorporated. That means there are results in the incorporated results mempool that are _not_ incorporated.

So I would suggest to 
1. Rename the `IncorporatedResults` back to `Results`
2. Rename the function `sealableResults` into `incorporatedResults`.
3. Change this [for-loop](https://github.com/onflow/flow-go/pull/41/files?diff=unified&w=1#diff-0834ce9a803801a4a1d3491c93529598R438) to be `for _, incorporatedResult := range incorporatedResults`, which would be more readable.

We can keep the name of `IncorporatedSeals` because those are seals for results that have passed the "IsIncorporated" check.